src/FakeItEasy/Core/AssemblyExtensions.cs

```diff
@@ -14,7 +14,16 @@ public static bool ReferencesFakeItEasy(this Assembly assembly)
         {
             Guard.AgainstNull(assembly, nameof(assembly));
 
+#if FEATURE_REFLECTION_GETASSEMBLIES
             return assembly.GetReferencedAssemblies().Any(r => r.FullName == TypeCatalogue.FakeItEasyAssembly.FullName);
+#else
+            var fakeItEasyLibraryName = TypeCatalogue.FakeItEasyAssembly.GetName().Name;
+            var context = Microsoft.Extensions.DependencyModel.DependencyContext.Default;
+
+            var runtimeLib = context.RuntimeLibraries
+                .SingleOrDefault(library => string.Equals(library.Name, assembly.Name(), System.StringComparison.Ordinal));

```

Comment:
> do you know why it's okay to equate a library name with an assembly name?

I couldn't find any docs either.  I might have modeled my code after some usage in ASP.NET repro, so there could be things that don't apply.  I will try to track down some docs/specs.

> I notice that a library has a list of assemblies in it. Should we be looking at those and comparing names instead of the library name?

Under debugger, they are all zero-length.  I don't know what they are for yet.